### PR TITLE
fix: use civilian payer for modified variants

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Ensure `computeFees()` run in modified transactions and queries by using a civilian payer for modified variants.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @tinker-michaelj

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java [hunk 1]
```diff
@@ -167,8 +175,16 @@ protected boolean submitOp(HapiSpec spec) {
                                                 : MAX_OPS_PER_SEC - opsThisSecond.get()))
                         .mapToObj(ignore -> provider.get())
                         .flatMap(Optional::stream)
-                        .filter(op -> op != Stream.empty())
-                        .peek(op -> counts.get(op.type()).getAndIncrement())
+                        .peek(op -> {
+                            counts.get(op.type()).getAndIncrement();
+                            if (loggingOff) {
+                                if (op instanceof HapiTxnOp<?> txnOp) {
+                                    txnOp.noLogging();
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java:182]: Why isn't `noLogging()` (and associated getter) part of parent `HapiSpecOperation`?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/QueryIdClearingStrategy.java [hunk 2]
```diff
@@ -27,26 +27,29 @@
...
+            Map.entry("proto.FileGetContentsQuery.fileID", ExpectedAnswer.onCostAnswer(INVALID_FILE_ID)),
+            Map.entry("proto.FileGetInfoQuery.fileID", ExpectedAnswer.onCostAnswer(INVALID_FILE_ID)),
             Map.entry(
                     "proto.TransactionID.accountID",
-                    ExpectedAnswer.onAnswerOnly(INVALID_ACCOUNT_ID, INVALID_TRANSACTION_ID)),
-            Map.entry("proto.ConsensusGetTopicInfoQuery.topicID", ExpectedAnswer.onAnswerOnly(INVALID_TOPIC_ID)),
-            Map.entry("proto.TokenGetInfoQuery.token", ExpectedAnswer.onAnswerOnly(INVALID_TOKEN_ID)),
-            Map.entry("proto.ScheduleGetInfoQuery.scheduleID", ExpectedAnswer.onAnswerOnly(INVALID_SCHEDULE_ID)),
-            Map.entry("proto.NftID.token_ID", ExpectedAnswer.onAnswerOnly(INVALID_TOKEN_ID)),
+                    new ExpectedAnswer(
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/QueryIdClearingStrategy.java:46]: This is the only place in the code where we have an `ExpectedAnswer` that satisfies both "cost answer" and "answer only".  What does it mean?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/SubmitModificationsOp.java [hunk 3]
```diff
@@ -67,4 +75,12 @@ protected boolean submitOp(@NonNull final HapiSpec spec) throws Throwable {
                         .toArray(HapiSpecOperation[]::new))));
         return false;
     }
+
+    private HapiTxnOp<?> originalTransaction() {
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/SubmitModificationsOp.java:79]: What does this method name mean when at L81-83 you actually change the payer?  Not "original" anymore?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/QueryModificationsOp.java [hunk 4]
```diff
@@ -33,29 +35,48 @@
 import java.util.stream.Stream;
 
 public class QueryModificationsOp extends UtilOp {
+    private static final String MODIFIED_CIVILIAN_PAYER = "modifiedCivilianPayer";
+    private final boolean useCivilianPayer;
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/QueryModificationsOp.java:39]: Please please add a comment about "civilian payer" per our discussion elsewhere!

**Review by david-bakin-sl — APPROVED** [review]
LGTM but there are comments that _could_ be addressed, if you would.  (E.g.,: Add a comment defining "civilian account" at the right place.)

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetContentsHandler.java [hunk 5]
```diff
@@ -82,7 +82,7 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         final var query = queryContext.query();
         final var fileStore = queryContext.createStore(ReadableFileStore.class);
         final var op = query.fileGetContentsOrThrow();
-        final var fileId = op.fileIDOrThrow();
+        final var fileId = op.fileIDOrElse(FileID.DEFAULT);
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetContentsHandler.java:85]: True, `findResponse()` can assume validity of any field already checked in `validate()` (such as this `fileID`).

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCallLocalHandler.java [hunk 6]
```diff
@@ -95,8 +95,9 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         // to call)
         final var contract = context.createStore(ReadableAccountStore.class).getContractById(contractID);
         if (contract == null) {
-            final var tokenID =
-                    TokenID.newBuilder().tokenNum(contractID.contractNum()).build();
+            final var tokenID = TokenID.newBuilder()
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCallLocalHandler.java:98]: Hmm...primitive integral types have default value `0` in both Java and protobuf spec, so seems everything should just work, right?

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountRecordsHandler.java [hunk 7]
```diff
@@ -117,9 +117,8 @@ public Fees computeFees(@NonNull final QueryContext queryContext) {
         final var query = queryContext.query();
         final var accountStore = queryContext.createStore(ReadableAccountStore.class);
         final var op = query.cryptoGetAccountRecordsOrThrow();
-        final var accountId = op.accountIDOrThrow();
+        final var accountId = op.accountIDOrElse(AccountID.DEFAULT);
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountRecordsHandler.java:120]: Created https://github.com/hashgraph/hedera-services/issues/13087

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/consensus/HapiGetTopicInfo.java [hunk 8]
```diff
@@ -205,7 +205,7 @@ protected boolean needsPayment() {
 
     @Override
     protected long lookupCostWith(HapiSpec spec, Transaction payment) throws Throwable {
-        Query query = getTopicInfoQuery(spec, payment, true);
+        Query query = maybeModified(getTopicInfoQuery(spec, payment, true), spec);
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/consensus/HapiGetTopicInfo.java:208]: Added javadoc to `maybeModified()`.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java:182]: Type inference for method chaining only works one level deep in the inheritance hierarchy; if we pull `noLogging()` up into `HapiSpecOperation`, then its return type would not be inferred as e.g. `HapiCryptoTransfer`.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/SubmitModificationsOp.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/SubmitModificationsOp.java:79]: Added javadoc.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/QueryModificationsOp.java (hunk 4 above)
Comment (tinker-michaelj) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/QueryModificationsOp.java:39]: Added.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/fees/calculation/file/queries/GetFileInfoResourceUsage.java [hunk 9]
```diff
@@ -64,7 +65,10 @@ public FeeData usageGiven(final Query query, final StateView view, final Map<Str
         return fileOpsUsage.fileInfoUsage(query, ctx);
     }
 
-    public FeeData usageGiven(final Query query, final File file) {
+    public FeeData usageGiven(final Query query, @Nullable final File file) {
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/fees/calculation/file/queries/GetFileInfoResourceUsage.java]: Added.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java (hunk 1 above)
Comment (david-bakin-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java:182]: Eeeee.... ok.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/SubmitModificationsOp.java [hunk 10]
```diff
@@ -39,12 +42,19 @@
  * original transaction.
  */
 public class SubmitModificationsOp extends UtilOp {
+    // An account to create as part of the submit-with-modifications process
+    // to ensure fees are charged (the default payer is a superuser account
+    // that is never charged fees); this ensures we cover calculateFees()
+    // code paths in handlers
+    private static final String MODIFIED_CIVILIAN_PAYER = "modifiedCivilianPayer";
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/mod/SubmitModificationsOp.java:49]: Unfortunately "civilian account" comment is also needed at `QueryModificationsOp`.  Sigh.  Don't change anything, but we really need that glossary.

**Review by david-bakin-sl — APPROVED** [review]
Unfortunately "civilian account" comment is also needed at `QueryModificationsOp`.  Sigh.  Don't change anything, but we really need that glossary.

**Pull request merged**